src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -0,0 +1,127 @@
+public class PopupService : IPopupService
+{
+	readonly IServiceProvider serviceProvider;
+
+	static readonly IDictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();
+
+	/// <summary>
+	/// Creates a new instance of <see cref="PopupService"/>.
+	/// </summary>
+	/// <param name="serviceProvider">The <see cref="IServiceProvider"/> implementation.</param>
+	public PopupService(IServiceProvider serviceProvider)
+	{
+		this.serviceProvider = serviceProvider;
+	}
+
+	internal static void AddTransientPopup<TPopupView, TPopupViewModel>(IServiceCollection services)
+		where TPopupView : Popup
+		where TPopupViewModel : INotifyPropertyChanged
+	{
+		viewModelToViewMappings.Add(typeof(TPopupViewModel), typeof(TPopupView));
+
+		services.AddTransient(typeof(TPopupView));
+		services.AddTransient(typeof(TPopupViewModel));
+	}
+
+	/// <inheritdoc cref="IPopupService.ShowPopup{TViewModel}()"/>
+	public void ShowPopup<TViewModel>() where TViewModel : INotifyPropertyChanged =>
+		ShowPopup(GetViewModel<TViewModel>());
+
+	/// <inheritdoc cref="IPopupService.ShowPopup{TViewModel}(TViewModel)"/>
+	public void ShowPopup<TViewModel>(TViewModel viewModel) where TViewModel : INotifyPropertyChanged
+	{
+		ArgumentNullException.ThrowIfNull(viewModel);
+
+		var popup = GetPopup(viewModel);
+
+		GetMainPage().ShowPopup(popup);
+	}
+
+	/// <inheritdoc cref="IPopupService.ShowPopup{TViewModel}(IDictionary{string, object})"/>
+	public void ShowPopup<TViewModel>(IDictionary<string, object> query) where TViewModel : IQueryAttributable
+	{
+		ArgumentNullException.ThrowIfNull(query);
+
+		var viewModel = GetViewModel<TViewModel>();
+		viewModel.ApplyQueryAttributes(query);
+
+		var popup = GetPopup(viewModel);
+
+		GetMainPage().ShowPopup(popup);
+	}
+
+	/// <inheritdoc cref="IPopupService.ShowPopupAsync{TViewModel}()"/>
+	public Task<object?> ShowPopupAsync<TViewModel>() where TViewModel : INotifyPropertyChanged =>
+		ShowPopupAsync(GetViewModel<TViewModel>());
+
+	/// <inheritdoc cref="IPopupService.ShowPopupAsync{TViewModel}(TViewModel)"/>
+	public Task<object?> ShowPopupAsync<TViewModel>(TViewModel viewModel) where TViewModel : INotifyPropertyChanged
+	{
+		ArgumentNullException.ThrowIfNull(viewModel);
+
+		var popup = GetPopup(viewModel);
+
+		return GetMainPage().ShowPopupAsync(popup);
+	}
+
+	/// <inheritdoc cref="IPopupService.ShowPopupAsync{TViewModel}(IDictionary{string, object})"/>
+	public Task<object?> ShowPopupAsync<TViewModel>(IDictionary<string, object> query) where TViewModel : IQueryAttributable
+	{
+		ArgumentNullException.ThrowIfNull(query);
+
+		var viewModel = GetViewModel<TViewModel>();
+		viewModel.ApplyQueryAttributes(query);
+
+		var popup = GetPopup(viewModel);
+
+		return GetMainPage().ShowPopupAsync(popup);
+	}
+
+	static Page GetMainPage()
+	{
+		var currentApplication = Application.Current;
+
+		ArgumentNullException.ThrowIfNull(currentApplication);
+
+		var mainPage = currentApplication.MainPage;
```

Comment:
This isn't the best way to get the current Page, because it can be the `NavigationPage` instead of the `ContentPage` presented on the screen.

I believe that you want the presented page that calls the popup. So, it's better to use this [method instead](https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Platform/PageExtensions.cs#L9)
We used it [here](https://github.com/communityToolkit/maui/blob/main/src/CommunityToolkit.Maui/Views/Popup/PopupExtensions.shared.cs#L88)  to find the page that called the popup and set it as its `Parent`.

Or you can try to find the page using the ViewModel

Reply:
Pedro helped me with FullScreen implementation of Page with:
```csharp
﻿namespace CommunityToolkit.Maui.Extensions;

// Since MediaElement can't access .NET MAUI internals we have to copy this code here
// https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Platform/PageExtensions.cs
static class PageExtensions
{
	internal static Page GetCurrentPage(this Page currentPage)
	{
		if (currentPage.NavigationProxy.ModalStack.LastOrDefault() is Page modal)
		{
			return modal;
		}
		if (currentPage is FlyoutPage fp)
		{
			return GetCurrentPage(fp.Detail);
		}
		if (currentPage is Shell shell && shell.CurrentItem?.CurrentItem is IShellSectionController ssc)
		{
			return ssc.PresentedPage;
		}
		if (currentPage is IPageContainer<Page> pc)
		{
			return GetCurrentPage(pc.CurrentPage);
		}

		return currentPage;
	}
}
```
Use this to get Page:
```csharp
protected Page CurrentPage =>
PageExtensions.GetCurrentPage(Application.Current?.MainPage ??
		                            throw new NullReferenceException("MainPage is null."));
```
And to actually get the page use this:
```csharp
Page page = CurrentPage;
```
I had no idea of how to get the current page in a generalized way. That code snippet will do it.